src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,187 @@
+                    }
+                    line = br.readLine();
+                }
+
+                this.count = list.size();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+    }
+
+    /**
+     * Constructor Method for the TaskList if no save file is given
+     */
+    public TaskList() {
+        list = new ArrayList<>();
+    }
+
+    /**
+     * @return ArrayList of tasks
+     */
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    /**
+     * Prints out the contents of the list
+     */
+    public String list() {
+        String s = "";
+        for (int i = 0; i < list.size(); i++) {
+            s += (i + 1 + ". " + list.get(i) + "\n");
+        }
+        return s;
+    }
+
+    /**
+     * Marks the content at index as done on the list
+     *
+     * @param index
+     */
+    public String mark(int index) {
+        String s = "";
+        list.get(index).mark();
+        s += "I have marked this task as done: \n";
+        s += list.get(index) + "\n";
+        return s;
+    }
+
+    /**
+     * Adds a new Event to the list
+     *
+     * @param e
+     */
+    public String add(Event e) {
+        String s = "";
+        try {
+            list.add(e);
+            count = list.size();
+            s += ("Added Task \n");
+            s += list.get(count - 1);
+            s += "Now you have " + count + " tasks in the list";
+            return s;
+        } catch (ArrayIndexOutOfBoundsException err) {
+            return "Invalid Time";
+        }
+    }
+
+    /**
+     * Adds a new Todo to the list
+     *
+     * @param t
+     */
+    public String add(Todo t) {
+        list.add(t);
+        count = list.size();
+        String s = "";
+        s += "Added Task \n";
+        s += list.get(count - 1) + "\n";
+        s += "Now you have " + count + " tasks in the list\n";
+        return s;
+    }
+
+    /**
+     * Adds a new Deadline to the list
+     *
+     * @param d
+     */
+    public String add(Deadline d) {
+        try {
+            list.add(d);
+            count = list.size();
+            String s = "";
+            s += "Added Task\n";
+            s += list.get(count - 1) + "\n";
+            s += "Now you have " + count + " tasks in the list" + "\n";
+            return s;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            return "Invalid Time";
+        }
+    }
+
+    /**
+     * Removes the task at index from the list
+     *
+     * @param index
+     */
+    public String delete(int index) {
+        Task item = list.get(index);
+        list.remove(index);
+        count = list.size();
+        String s = ("Noted. I have removed this task: \n" + item);
+        return s;
+    }
+
+    /**
+     * Marks the task at index as not done
+     *
+     * @param index
+     */
+    public String unMark(int index) {
```

Comment:
```suggestion
     public String unmark(int index) {
```
'unmark' is a single word so the 'M' should not be capitalized.